app/models/katello/activation_key.rb

```diff
@@ -177,6 +177,17 @@ def self.find_by_purpose_addons(_key, operator, value)
       return_activation_keys_by_id(activation_keys.pluck(:id))
     end
 
+    def validate_destroyable!
+      unless organization.being_deleted?
+        if Parameter.where(name: 'kt_activation_keys').pluck(:value).any? { |value| value.split(",").include?(name) }
+          fail _("This activation key is associated to one or more Hosts/Hostgroups. "\
+                  "Search and unassociate Hosts/Hostgroups using params.kt_activation_keys ~ \"%{name}\" "\
```

Comment:
How/where should the users do the search and disassociation?

Reply:
For the record, `params.kt_activation_keys ~ "key1"` works in the hostgroups'  search box.